pennylane/qnode.py

```diff
@@ -411,11 +414,17 @@ def _validate_adjoint_method(device, interface):
                 f"The {device.short_name} device does not support adjoint differentiation."
             )
 
+        jac_options = {"method": "device", "jacobian_method": "adjoint_jacobian"}
+        # reuse the forward pass
+        # torch and tensorflow can cache the state
+        if interface in {"autograd", "jax"}:
+            jac_options["device_pd_options"] = {"use_device_state": True}

```

Comment:
I'm confused on the difference between the `use_device_state` workflow and the `starting_state` workflow.

So we have:
- Autograd and jax use the device state
- Torch and TF use `starting_state`, where `starting_state` is the device's state but loaded from the forward pass within the interface

Can autograd and jax not use `starting_state`? Can Torch and TF not use the device state approach?

Reply:
I don't want autograd and jax to use `starting_state`, as that requires more moving around of large arrays.  Torch and TF can't use the device state, because we might perform the backward pass after multiple forward passes.